# About Project
Supplemental seqeunce processing and R statistical analysis for phytobiomes publication which compares the microbiome of 12 Zea genotypes, 6 modern maize and 6 wild teosinte. 

# Sample Data 
Raw sample data is present at BioProject: [PRJNA753114](https://urldefense.com/v3/__https://dataview.ncbi.nlm.nih.gov/object/PRJNA753114?reviewer=76e2fve465rj6t7oku2e30ihf__;!!DZ3fjg!pmcDWRcgjR62Vx5fJu8tTdBCgGB006RZu8Y-cUtWZ_qU36IiJYFlDQlP21xu5klxkSw$)

# Collaborators 
1. [Angela Kent](https://microbiome.nres.illinois.edu/)
2. [Martin Bohn](https://cropsciences.illinois.edu/people/profile/mbohn)

# Tools Used 
Phase 1 describes bioinfromatics and phase 2 describes statisitical analysis. 
------------------
Phase 1 - Flash for merging R1, R2 fastq - Link
Phase 1 - Fastx toolkit for quality filtering - Link 
Phase 1 - Usearch for OTU construction - Link
Phase 1 - QIIME for OTU Table - Link
Phase 2 - Analysis in R and R Studio - Link 
Phase 2 - Phyloseq for visualization of amplicon - Link 
Phase 2 - ggplot2 for figure plotting - Link 
Phase 2 - Vegan for multivariate stats - Link 
Phase 2 - DESEQ2 for Differential Abundance - Link
Phase 2 - ASREML-R for Univariate stats- Link
